modules/calib3d/src/usac/estimator.cpp

```diff
@@ -334,17 +336,16 @@ ReprojectionErrorForward::create(const Mat &points) {
 
 class SampsonErrorImpl : public SampsonError {
 private:
-    const Mat * points_mat;
-    const float * const points;
+    Mat points_mat;
     float m11, m12, m13, m21, m22, m23, m31, m32, m33;
     std::vector<float> errors;
 public:
     explicit SampsonErrorImpl (const Mat &points_)
-        : points_mat(&points_), points ((float *) points_.data)
+        : points_mat(points_)
         , m11(0), m12(0), m13(0), m21(0), m22(0), m23(0), m31(0), m32(0), m33(0)
         , errors(points_.rows)
     {
-        CV_DbgAssert(points);
+        CV_DbgAssert(!points_mat.empty());
```

Comment:
I think we should also add `CV_Assert(points.isContinuous())` check. Here and in other places where Mat objects enter the scope. Maybe type and dimensions should be verified too?

Reply:
Done!